### PR TITLE
Add onDismiss function option when a toast is created

### PR DESCRIPTION
**Pull request by doncicuto**
This component is quite useful, thanks a lot! 

In my case, I wanted to perform some actions that should only be triggered once the toast was explicitely dismissed by clicking the close sign, so I've added the onDismiss support. onClose it is called either you click on the close sign or if the time passes and the toast is closed. With the onDismiss function, if you want to do specific things when the X is clicked, you now can do your stuff before the toast is closed.

### CODE REVIEW

File: lib/semantic-toast.jsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ function SemanticToast(props) {
 
     const onDismiss = e => {
         e.stopPropagation();
+        props.onDismiss();
```
Comment (ferrao) [inline lib/semantic-toast.jsx]: Can you please destructure `onDismiss` as well?

**Comment by ferrao** [comment]
Don't understand the title of this PR, does not seem related to your description/code. Can you please clarify?

File: lib/semantic-toast.jsx (hunk 1 above)
Comment (doncicuto) [reply lib/semantic-toast.jsx]: Done!

**Comment by michael-harley** [comment]
@ferrao @doncicuto I got onDismiss is not a function whenever I hit the close button, and my toast is not closing. (Can only wait until it's time end)
I think this PR is the cause?

image

**Comment by michael-harley** [comment]
@ferrao @doncicuto This change is not applied to npmjs, or is it just me?

I tried ` "react-semantic-toasts": "^0.6.4"` for my package.json, but I only got this code for `semantic-toast.jsx`.
image

This is causing an error `onDismiss is not a function`.
image

-----
I can easily solve this by changing my package.json into `react-semantic-toasts: "https://github.com/academia-de-codigo/react-semantic-toasts"`, but this is not elegant and I want my package.json to be clean.

Can you help me with this, maybe I do something wrong.

-----
**I've tried:**
- Remove node_modules and yarn.lock
- Re-run `yarn` command.

**Comment by doncicuto** [comment]
Hi @michael-harley,
I've created a small React app with create-react-app and I've tried to download react-semantic-toasts (0.6.4) from npmjs using yarn. onDismiss seems to work and I've uploaded the code to this repo: https://github.com/doncicuto/test-toast 

Captura de pantalla de 2019-11-05 18-12-04

I've only set size to small as it complained about medium size not defined in current React Semantic version which has already been tackled in @JanKaifer PRs.

I'm not sure what may be the cause for your issue but if I can help you let me know.

Cheers

**Pull request merged**